stateresolution.go

```diff
@@ -41,12 +41,17 @@ func ResolveStateConflicts(conflicted []Event, authEvents []Event) []Event {
 //
 // It implements the AuthEvents interface and can be used for running auth checks.
 type stateResolver struct {
-	creates                   []Event
-	powerLevels               []Event
-	joinRules                 []Event
-	thirdPartyInvites         [][]Event
-	members                   [][]Event
-	others                    [][]Event
+	// Lists of lists of events to resolve grouped by event type and state key:
+	//   * creates, powerLevels, joinRules have empty state keys.
+	//   * members and thirdPartyInvites are grouped by state key.
+	//   * the others are grouped by type and state key.
```

Comment:
The `others` then feels like it should be a 3d-array based on this description but it isn't, presumably because the "key" is the tuple of (event type, state key) already. Perhaps word it as:
```go
// List of lists of events to resolve grouped by the tuple of (event type, state key):
// * creates, powerLevels and joinRules have 0-length state keys so don't
//   need to be grouped by the tuple.
// * members, thirdPartyInvites and others are grouped by the tuple: each
//   element in the outer array represents a specific tuple, and each Event in the inner
//   array is guaranteed to have the same type/state key as each other.
```